src/MSALWrapper/GlobalLock.cs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Authentication.MSALWrapper
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// GlobalLock is used for preventing code from entering critical sections in different processes at the same time.
+    /// </summary>
+    public class GlobalLock : IDisposable
+    {
+        private Mutex mutex;
+        private bool disposedValue;
+        private bool lockAcquired;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="GlobalLock"/> class.
+        /// </summary>
+        /// <param name="lockName">Unique lock ID.</param>
+        /// <exception cref="ArgumentNullException">The <see cref="ArgumentNullException"/>.</exception>
+        public GlobalLock(string lockName)
+        {
+            if (lockName == null)
+            {
+                throw new ArgumentNullException($"{nameof(lockName)} must not be null");
+            }
+
+            this.mutex = new Mutex(false, lockName);
+            try
+            {
+                this.mutex.WaitOne();

```

Comment:
Again, here most people aren't familiar with the Mutex API, so a short comment here on why we are calling `WaitOne()` would be nice.

Reply:
I agree with @kyle-rader here. Additionally, there's no timeout associated with this, so we might end up waiting forever. I'm not sure that's what we want to do.